### DISCUSSION TITLE
Tissue mean - LSQ

### DISCUSSION THREAD

**Discussion by RaissaSouza** [jcreinhold/intensity-normalization#65] [Q&A] [answered]
Hello Jacob Reinhold, 
I am using your python intensity-normalization code to harmonize my data (Nifit). When applying the least square normalization method, I noticed that the tissue mean only has 2 values. I think this indicates a problem in my data since the code could not identify WM, GM, CSF. Am I right?  Could you clarify for me what is suppose to be saved in the tissue means values. Is there any documentation, pre-print for LSQ, I did not find on your paper?

**Comment by jcreinhold** [comment, marked as answer]
How are you checking that the tissue mean only has two values? Can you provide the code you're running and the output?

**Reply by RaissaSouza** [reply]
Sure this is the way I am running it:

```python
def lsq_harmonization_fit(images, masks,outDir):
  lsq_normalizer = LeastSquaresNormalize()
  lsq_normalizer.fit(images,masks)
  lsq_normalizer.save_standard_tissue_means(outDir+"tissue_means.npy")

  
def lsq_harmonization_apply(images, masks,outDir):
  lsq_normalizer = LeastSquaresNormalize()
  lsq_normalizer.load_standard_tissue_means(outDir+"tissue_means.npy")
  normalized = [lsq_normalizer(image,mask) for image, mask in zip(images,masks)]
  return normalized
```

Before calling the apply I just printed the values of the tissue mean this way:

```python
data = np.load("tissue_means.npy")
standard_scale = data[0, :]
percentiles = data[1, :]
print(standard_scale)
print(percentiles)
```

This is the output:
```
[1.00000003]
[2.4359278]
```
And I was actually wrong, there is one value for the scale and 1 for percentiles, and I am confused how only this 2 values are able to identify GM, WM, CSF.

**Reply by jcreinhold** [reply]
I'm guessing you looked at the Nyul code for the "standard scale" and "percentiles". LSQ works differently and doesn't use either. If you just print `data`, i.e., `print(data)` (using the variable names from your code) you'll see the three values where the first entry corresponds to CSF, the second to GM, and the third to WM.

**Reply by RaissaSouza** [reply]
You are totally right! I messed up the codes. Thanks for the clarification ;)
